homeassistant/components/tractive/sensor.py

```diff
@@ -0,0 +1,178 @@
+"""Support for Tractive sensors."""
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import (
+    ATTR_BATTERY_LEVEL,
+    DEVICE_CLASS_BATTERY,
+    PERCENTAGE,
+    TIME_MINUTES,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+
+from .const import (
+    ATTR_DAILY_GOAL,
+    ATTR_MINUTES_ACTIVE,
+    DOMAIN,
+    SERVER_UNAVAILABLE,
+    TRACKER_ACTIVITY_STATUS_UPDATED,
+    TRACKER_HARDWARE_STATUS_UPDATED,
+)
+from .model import TractiveSensorEntityDescription
+
+ATTR_ACTIVITY = "activity"
+ATTR_HARDWARE = "hardware"
+
+SENSOR_TYPES = (
+    TractiveSensorEntityDescription(
+        key=ATTR_BATTERY_LEVEL,
+        name="Battery Level",
+        native_unit_of_measurement=PERCENTAGE,
+        device_class=DEVICE_CLASS_BATTERY,
+        event_type=ATTR_HARDWARE,
+    ),
+    TractiveSensorEntityDescription(
+        key=ATTR_MINUTES_ACTIVE,
+        name="Minutes Active",
+        icon="mdi:clock-time-eight-outline",
+        native_unit_of_measurement=TIME_MINUTES,
+        event_type=ATTR_ACTIVITY,
+        attributes=(ATTR_DAILY_GOAL,),
+    ),
+)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up Tractive device trackers."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    trackables = await client.trackable_objects()
+
+    entities = []
+
+    for item in trackables:
+        trackable = await item.details()
+        tracker = client.tracker(trackable["device_id"])
+        tracker_details = await tracker.details()
+        for description in SENSOR_TYPES:
+            unique_id = f"{trackable['_id']}_{description.key}"
+            if description.event_type == ATTR_HARDWARE:
+                entities.append(
+                    TractiveHardwareSensor(
+                        client.user_id,
+                        trackable,
+                        tracker_details,
+                        unique_id,
+                        description,
+                    )
+                )
+            if description.event_type == ATTR_ACTIVITY:
+                entities.append(
+                    TractiveActivitySensor(
+                        client.user_id,
+                        trackable,
+                        tracker_details,
+                        unique_id,
+                        description,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class TractiveSensor(SensorEntity):
```

Comment:
This is probably a great time to add a shared  `TractiveEntity` with the attributes that both platforms use.